AssignClassProps

aljgardin opened the pull request:
Added AssignClassProps to Mapper classes that needed it.
Added missing properties to FPC: ReadSchema, WriteSchema and WriteProject.

graemeg committed:
Restructure

aljgardin committed:
Merge branch 'master' of https://github.com/aljgardin/tiopf_apps

aljgardin committed:
Merge branch 'master' of https://github.com/aljgardin/tiopf_apps

aljgardin committed:
Merge branch 'master' of https://github.com/aljgardin/tiopf_apps

aljgardin committed:
Adds

aljgardin committed:
Adds

aljgardin committed:
Merge branch 'master' of https://github.com/aljgardin/tiopf_apps

# Conflicts:
#	tiOPFMapper/utils/lazarus/Dialogs/frmagtiautoeditdialog.lfm
#	tiOPFMapper/utils/lazarus/Dialogs/frmagtiautoeditdialog.pas

aljgardin committed:
Merge branch 'master' of https://github.com/aljgardin/tiopf_apps

# Conflicts:
#	tiOPFMapper/utils/lazarus/Dialogs/frmagtiautoeditdialog.lfm
#	tiOPFMapper/utils/lazarus/Dialogs/frmagtiautoeditdialog.pas

aljgardin committed:
Merge branch 'master' of https://github.com/aljgardin/tiopf_apps

# Conflicts:
#	tiOPFMapper/MGPropertyDataGUI/MGPropertyDataGUI.xml
#	tiOPFMapper/utils/lazarus/frmmgmain.lfm
#	tiOPFMapper/utils/lazarus/frmmgmain.pas
#	tiOPFMapper/utils/lazarus/lib/i386-win32/frmmgmain.lfm
#	tiOPFMapper/utils/lazarus/testform.lfm

aljgardin committed:
Delete mapper_Base.pas

aljgardin committed:
Delete config.xml

aljgardin committed:
Delete tbatch.bat

aljgardin committed:
Delete timapper.xml

aljgardin committed:
Delete Mapper.inc

aljgardin committed:
Delete MapperClasses.txt

aljgardin committed:
Delete timapper.xml

aljgardin committed:
Major changes.

aljgardin committed:
Major Changes and restructure of my files.

aljgardin committed:
Changes to Mapper.Pas:

Added Forward declares for some new classes.
	New Classes are MG prefixed for Mapper GUI.

TMapProject:
	Added:
		procedure AssignClassProps
		property IncludeFiles: TMGIncludeList
			a Mediator friendly Includes list.
		procedure OnIncludesChange
			When Includes stringlist changes, update IncludeFiles.
TPropMapping:
	Added:
		Property FieldLength.

TMapClassDef:
	Added:
		property DirttyOnPropertyChange

TMapUnitDef:
	Added:
		property UnitReferences: TMGReferenceList
			a Mediator friendly References list.
		procoedrue AssignClassProps

Added AssignClassProps to:
	TMapEnum,
	TClassMapping,
	TClassMappingSelect,
	TFilterDef,

	New Classes:
	  TMGInclude = Class(TBaseMapObject)
	  TMGIncludeList = class(TBaseMapObjectList)
	  TMGReference = class(TBaseMapObject)
	  TMGReferenceList = class(TBaseMapObjectList)
New Functions:
	function gClassDefTypeToStr(const AClassDefType: TClassDefType): String;
	function gOIDTypeToStr(const AOIDType: TOIDType): String;

aljgardin committed:
Cleanup files.

aljgardin committed:
Added AssignClassProps to All mapper classes that needed it.
Added ReadSchema, WriteSchema for missing properties of MapProject.
Added Missing properties to WriteProject.

aljgardin committed:
Delete ajgmapperchanges.txt

aljgardin committed:
Delete Modelwithenums.pas

aljgardin committed:
Delete Modelwithenums.pas

aljgardin commented:
Still getting the hang of Git.

aljgardin committed:
Merge branch 'master' of https://github.com/aljgardin/tiopf_apps

graemeg commented:
I really appreciate the efforts to contribute, but unfortunately you didn't base your pull request on the latest code from my repository. So if I had to apply your pull request, it will actually undo a whole lot of my recent changes. To make things easier for yourself (and for me), always do your modifications in a separate branch. That way you can rebase or branch on the latest code from my repository - as my repository moves forward. Also, such a pull requests will only show your changes (without undoing my recent changes).

So if you would like to do the above and try another pull request I'll happily look at it again. If you need more guidance with Git, don't hesitate to as in the tiopf.support list. I can help were possible, or give you good link to web pages that describe the process.

graemeg commented:
To add to my previous comment... If you use a separate branch per "feature" you introduce, you also don't need to add lines like the following in your code:

`{AJG: ADDED:}`

Because with Git and a separate branch for changes, I'll clearly be able to see the lines of code you changed, added or deleted.

aljgardin commented:
sorry, I forked your main.Should I branch from my fork or branch from your main?I was using the {AJG: ADDED:} for me to find my changes.
 Albert J Gardin

 

    On Saturday, July 22, 2017 1:25 PM, Graeme Geldenhuys <notifications@github.com> wrote:
 

 To add to my previous comment... If you use a separate branch per "feature" you introduce, you also don't need to add lines like the following in your code:{AJG: ADDED:}Because with Git and a separate branch for changes, I'll clearly be able to see the lines of code you changed, added or deleted.—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

graemeg commented:
No worries. When you clone my repository, you will end up in your local "master" branch which is the same as my repo's master branch. You then simply create a feature branch, commit changes into there, then publish that feature branch to your GitHub account.

Working with branches has many benefits when using Git. To see your change you simply need to do a "git diff origin/master..<feature_branch_name>" and that will show you all your changes with out needing the {AJG: xxx} comments.

graemeg commented:
This pull request was replaced with a new and improved one.
